src/helpers.rs

```diff
@@ -150,3 +151,68 @@ where
 pub(crate) fn path_from_null_terminated(bytes: Vec<u8>) -> PathBuf {
     Path::new(OsStr::from_bytes(bytes.split_last().unwrap().1)).to_path_buf()
 }
+
+pub(crate) fn is_snap() -> bool {
+    let pid = std::process::id();
+    let path = format!("/proc/{}/cgroup", pid);
+    let mut file = match std::fs::File::open(path) {
+        Ok(file) => file,
+        Err(_) => return false,
+    };
+
+    let mut buffer = String::new();
+    match file.read_to_string(&mut buffer) {
+        Ok(_) => parse_cgroup_v2(&buffer),
+        Err(_) => false,
+    }
+}
+
+fn parse_cgroup_v2(cgroups: &str) -> bool {
```

Comment:
Ideally this returns a Option<None> but it is a bit hard to do so with the current logic, we don't want to return None if the first line is not what we want.

Reply:
The method itself is a bit weird, it is called parse_cgroup_v2 but only checks if it is a snap app so. Maybe call it cgroup_v2_is_snap or something like that.